greenmail-core/src/main/java/com/icegreen/greenmail/configuration/PropertiesBasedGreenMailConfigurationBuilder.java

```diff
@@ -64,4 +125,16 @@ protected void extractAndAddUser(GreenMailConfiguration configuration, String us
                         + " parsed to " + Arrays.toString(userParts));
         }
     }
+
+    /**
+     * Parses a single user in the format logon:password[@domain] where the domain part is optional. Valid
+     * returned Strings arrays have 2 elements (without domain) or 3 elements (with domain).
+     *
+     * @param user - the user in format logon:password[@domain]
+     * @return a String array with the different parts, either of length 2 or 3
+     */
+    private static String[] parseUser(String user) {
+        return user.trim().split(":|@");
```

Comment:
I completely agree, but I did not want to refactor existing code too much, I wanted to concentrate on the new file-store-functionality.

==> But I will change it using regex with capturing groups.